lisp/forge-gitlab.el

```diff
@@ -542,6 +542,16 @@ it is all or nothing.")
   (--filter (string-match-p "\\`\\.gitlab/merge_request_templates/.+\\.md\\'" it)
             (magit-revision-files (oref repo default-branch))))
 
+(cl-defmethod forge--fork-repository ((repo forge-gitlab-repository) fork)
+  (with-slots (owner name) repo
+    (let ((resource (format "/projects/%s%%2F%s" fork name)))
+      (if (glab-get resource)
+          (forge--msg nil t nil "Repo %s/%s already exists!" fork name)
+        (forge--glab-post repo (format "/projects/%s%%2F%s/fork" owner name)
```

Comment:
The github variant intentionally does not check whether the fork already exists. That way `forge-fork` can be (ab)used to add a remote for a certain fork regardless of whether that fork already exists (and the user having to remember whether that is so or not).